### PR TITLE
fix: Change currency of ad-generation's bidResponse according to bidRequest

### PR DESCRIPTION
**Pull request by AdGeneration**
## Type of change
fix the following contents
https://github.com/prebid/prebid-server/pull/1253#discussion_r448985371

## Description of change
Change currency of ad-generation's bidResponse according to bidRequest

Maintainer: ssp-ope@supership.jp

### CODE REVIEW

**Comment by SyntaxNode** [comment]
Change looks good. Can you please add a unit test for it?

**Comment by AdGeneration** [comment]
@SyntaxNode  @bsardo

Thank you for reviewing.
I added a test code, so please check it.

In addition to the changes above, we also changed the version for the external bid request.

File: adapters/adgeneration/adgeneration_test.go [hunk 1]
```diff
@@ -174,3 +175,82 @@ func TestCreateAd(t *testing.T) {
...
... \"displaytype\": \"1\",\n \"h\": 250,\n \"ids\": {\n \"anid\": \"\",\n \"diid\": \"\",\n \"idfa\": \"\",\n \"soc\": \"Soc\"\n },\n \"location_params\": null,\n \"locationid\": \"58278\",\n \"results\": [\n {\n \"ad\": \"<div id=\\\"medibasspContainer\\\">\\n<iframe src=\\\"https://dummy-iframe.com></iframe>\\n</div>\",\n \"beacon\": \"<img src=\\\"https://dummy-beacon.com\\\">\",\n \"beaconurl\": \"ttps://dummy-beacon.com\",\n \"cpm\": 30,\n \"creativeid\": \"Dummy_supership.jp\",\n \"h\": 250,\n \"landing_url\": \"\",\n \"scheduleid\": \"512601\",\n \"trackers\": {\n \"imp\": [\n \"https://tg.socdm.com/bc/v3?b=beaconID.gif\"\n ],\n \"viewable_imp\": [\n \"https://tg.socdm.com/aux/inview\"\n ],\n \"viewable_measured\": [\n \"https://tg.socdm.com/aux/measured\"\n ]\n },\n \"ttl\": 10,\n \"w\": 300,\n \"weight\": 1\n }\n ],\n \"dealid\": \"test-deal-id\",\n \"rotation\": \"0\",\n \"scheduleid\": \"512601\",\n \"sdktype\": \"0\",\n \"trackers\": {\n \"imp\": [\n \"https://tg.socdm.com/bc/v3?b=beaconID.gif\"\n ],\n \"viewable_imp\": [\n \"https://tg.socdm.com/aux/inview\"\n ],\n \"viewable_measured\": [\n \"https://tg.socdm.com/aux/measured\"\n ]\n },\n \"ttl\": 10,\n \"w\": 300\n}"),
```
Comment (SyntaxNode) [inline adapters/adgeneration/adgeneration_test.go:193]: For the purposes of this unit test, could you please use a simplified  request? For example, the "ad" could just me "testAd", and the "adM" could be "testAdM". This works, but is difficult to read. You can also use the json files to represent these two tests cases which is designed for improved readability as well.

File: adapters/adgeneration/adgeneration_test.go [hunk 2]
```diff
@@ -174,3 +175,82 @@ func TestCreateAd(t *testing.T) {
...
+	var expectedH uint64 = 250
+	var expectedCrID string = "Dummy_supership.jp"
+	var extectedDealID string = "test-deal-id"
+
+	// Currency
+	if bidderRespoonse.Currency != expectedCurrency {
+		t.Errorf("Currency %v does not equal %v", bidderRespoonse.Currency, expectedCurrency)
+	}
+	// Bids
+	if len(bidderRespoonse.Bids) != 1 {
```
Comment (SyntaxNode) [inline adapters/adgeneration/adgeneration_test.go:229]: If you don't move to json tests, consider using the testify package to simplify these lines to:
```
assert.Equal(t, 1, len(bidderRespoonse.Bids), "Bid Responses")
assert.Equal(t, expectedID, bidderRespoonse.Bids[0].Bid.ID, "Bid ID")
...
```
The  testify package will auto generate useful error messages.

**Comment by AdGeneration** [comment]
Thank you for reviewing.
Also, I am sorry that the correction was delayed.

I fixed two points that you have reviewed.

> could you please use a simplified request?

Especially this part may be misunderstood by me, so it would be helpful if you could check it.

**Comment by SyntaxNode** [comment]
> @SyntaxNode
> Thank you for reviewing.
> Also, I am sorry that the correction was delayed.
> 
> I fixed two points that you have reviewed.
> 
> > could you please use a simplified request?
> 
> Especially this part may be misunderstood by me, so it would be helpful if you could check it.

Thank you. I was trying to see if the sample adM in the test could be reduced from `"<div>\n<iframe></iframe>\n</div>\n` to something like `anyAdM` to keep it easier to read.

**Comment by AdGeneration** [comment]
Thank you for explaining.
I fixed it. Please confirm.

**Comment by SyntaxNode** [comment]
Looks great. Thank you.

**Pull request merged**